src/evm/StateMachine.ts

```diff
@@ -0,0 +1,164 @@
+import { ethers, BytesLike, hexlify } from "ethers";
+import { TransactionStruct } from "@typechain-types/contracts/V1/DataTypes";
+import { ContractExecuter } from "./ContractExecuter";
+import { EVM, Log } from "@ethereumjs/evm";
+import { Address } from "@ethereumjs/util"
+/**
+ * Provides a typed interface to interact with a state machine contract
```

Comment:
Don't see the need for this file and it's confusing since the EvmStateMachine is the state machine that's used polymorphically (implements AStateMachine) so this is just confusing and doesn't do anything special - I do understand that the EvmStateMachine wasn't refactored and that some functionality could be wrapped into a function and reused (that's good) but this file is not needed - I would refactor the few useful functions within the EvmStateMachine and avoid the confusion of multiple stateMachine files (even our contracts have these names)

Reply:
removed